api/api.go

```diff
@@ -78,48 +78,91 @@ type API struct {
 	maxMergeBatchSize int64
 	targets           func(context.Context) TargetRetriever
 	globalURLOptions  GlobalURLOptions
+	prefix            string
 
 	mu     sync.RWMutex
 	config *config.Config
 }
 
+type Option func(*API)
+
 func New(
 	logger log.Logger,
 	registry *prometheus.Registry,
-	db storage.Queryable,
-	reloadCh chan struct{},
-	maxMergeBatchSize int64,
-	targets func(ctx context.Context) TargetRetriever,
+
+	opts ...Option,
 ) *API {
-	return &API{
-		logger:            logger,
-		registry:          registry,
-		db:                db,
-		reloadCh:          reloadCh,
-		maxMergeBatchSize: maxMergeBatchSize,
-		targets:           targets,
+	a := &API{
+		logger:   logger,
+		registry: registry,
+		prefix:   "/api/v1/",
+		reloadCh: make(chan struct{}),
 		globalURLOptions: GlobalURLOptions{ // TODO pass into from flags
```

Comment:
Side note: Ah maybe this could be a nice first issue for someone to do, what do you if we open an issue for this?